tensorflow/compiler/xla/shape_tree.h

```diff
@@ -653,14 +653,15 @@ void ShapeTree<T>::CopySubtreeFrom(const ShapeTree<T>& other,
                             const ShapeIndex& index, T* data) {
     // Copy the data element only if index is in the
     // subtree rooted at target_base_index.
-    for (int i = 0; i < target_base_index.size(); ++i) {
-      if (i >= index.size() || index[i] != target_base_index[i]) {
+    const int index_size = index.size();
+    for (int i = 0, end = target_base_index.size(); i < end; ++i) {
+      if (i >= index_size || index[i] != target_base_index[i]) {
         return;
       }
     }
     // Construct source element index to copy from.
     ShapeIndex source_index = source_base_index;
-    for (int i = target_base_index.size(); i < index.size(); ++i) {
+    for (int i = target_base_index.size(), end = index.size(); i < end; ++i) {
```

Comment:
Here too, changing type of `i` should be enough. Plus, loop can be written as a range-loop instead

Reply:
updated